Tpetra TSQR:  shadow warning

## Bug Report
@trilinos/tpetra 

### Description
TSQR produces a shadow warning with gcc 6.4.
```
Trilinos/packages/tpetra/tsqr/src/Tsqr.hpp:327:55: warning: declaration of 'R_view' shadows a previous local [-Wshadow]
         mat_view_type R_view (numCols, numCols, R, LDR);
                                                       ^
Trilinos/packages/tpetra/tsqr/src/Tsqr.hpp:269:21: note: shadowed declaration is here
       mat_view_type R_view (numCols, numCols, R, LDR);

```

### Steps to Reproduce
```
module load sems-env
module load sems-gcc/6.4.0
module load sems-openmpi/1.10.1

cmake \
-D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
-D CMAKE_BUILD_TYPE:STRING="DEBUG" \
\
-D MPI_BIN_DIR:PATH="${SEMS_MPI_ROOT}/bin" \
-D TPL_ENABLE_MPI:BOOL=ON \
-D MPI_EXEC_MAX_NUMPROCS:STRING=11 \
\
-D TPL_ENABLE_BinUtils:BOOL=OFF \
-D TPL_ENABLE_Pthread:BOOL=OFF \
\
-D CMAKE_C_FLAGS:STRING="-Wall  -pedantic -Wno-unknown-pragmas -Wno-narrowing -Wno-inline -Wshadow -Wdeprecated-declarations -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -fstrict-aliasing -Wno-long-long" \
-D CMAKE_CXX_FLAGS:STRING="-Wall -pedantic -Wno-unknown-pragmas -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-inline -Wshadow -Wdeprecated-declarations -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -Wvla -fstrict-aliasing" \
\
-D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
-D Trilinos_ENABLE_TESTS:BOOL=OFF \
-D Trilinos_ENABLE_EXAMPLES:BOOL=OFF \
\
-D Trilinos_ENABLE_SHADOW_WARNINGS:BOOL=ON \
-D Trilinos_VERBOSE_CONFIGURE:BOOL=OFF \
-D Trilinos_ENABLE_Fortran:BOOL=OFF \
\
-D Trilinos_ENABLE_Epetra:BOOL=ON \
-D Trilinos_ENABLE_Galeri:BOOL=ON \
-D Trilinos_ENABLE_Pamgen:BOOL=ON \
\
-D Trilinos_ENABLE_Anasazi:BOOL=ON \
-D Anasazi_ENABLE_EXAMPLES:BOOL=ON \
-D Anasazi_ENABLE_TESTS:BOOL=ON \
-D Anasazi_ENABLE_TSQR:BOOL=ON \
-D Trilinos_ENABLE_Tpetra:BOOL=ON \
-D Tpetra_ENABLE_EXAMPLES:BOOL=ON \
-D Tpetra_ENABLE_TESTS:BOOL=ON \
\
-D Teuchos_ENABLE_STACKTRACE:BOOL=OFF \
-D Teuchos_ENABLE_LONG_LONG_INT:BOOL=ON \

--- a/packages/tpetra/tsqr/src/Tsqr.hpp
+++ b/packages/tpetra/tsqr/src/Tsqr.hpp
@@ -249,7 +249,7 @@ namespace TSQR {
                        const bool forceNonnegativeDiagonal = false)
     {
       const char prefix[] = "TSQR::Tsqr::factorExplicitRaw: ";
-      
+
       // Sanity checks for matrix dimensions.
       if (numRows < numCols) {
         std::ostringstream os;
@@ -289,7 +289,7 @@ namespace TSQR {
           (true, std::runtime_error, prefix <<
            "nodeTsqr_->factor(...) threw: " << e.what ());
       }
-      
+
       // Prepare the output matrix Q by filling with zeros.
       try {
         nodeTsqr_->fill_with_zeros (numRows, numCols, Q, LDQ,
@@ -300,7 +300,7 @@ namespace TSQR {
           (true, std::runtime_error, prefix <<
            "nodeTsqr_->fill_with_zeros(...) threw: " << e.what ());
       }
-      
+
       // Wrap the output matrix Q in a "view."
       mat_view_type Q_rawView (numRows, numCols, Q, LDQ);
       // Wrap the uppermost cache block of Q.  We will need to extract
@@ -324,7 +324,6 @@ namespace TSQR {
       {
         mat_view_type Q_top (numCols, numCols, Q_top_block.data (),
                              Q_top_block.stride (1));
-        mat_view_type R_view (numCols, numCols, R, LDR);
 
         if (nodeTsqr_->wants_device_memory ()) {
           // DistTsqr doesn't know what to do with device memory, so

